### PR TITLE
fix bugs found when enabling retries for all tests

### PR DESCRIPTION
**Pull request by markdroth**
Pulling this out of #26766.

This includes a couple of fixes needed to get all tests to pass when retries are enabled by default:
- Change retry filter to properly propagate some missing op payload fields.
- Fix http_client filter to restore the send_message op to the batch before returning completion of the batch, so that the retry code knows which ops were in that batch.

### CODE REVIEW

**Comment by markdroth** [comment]
Known issues: #26595

File: src/core/ext/filters/client_channel/retry_filter.cc [hunk 1]
```diff
@@ -1593,6 +1596,10 @@ void RetryFilter::CallData::CallAttempt::BatchData::
     call_attempt_->recv_trailing_metadata_error_ = error;
     return;
   }
+  // Copy transport stats to be delivered up to the surface.
```
Comment (yashykt) [inline src/core/ext/filters/client_channel/retry_filter.cc:1599]: Note that this is a `move`, not a `copy`

**Review by yashykt — APPROVED** [review]
approved with a nit

File: src/core/ext/filters/client_channel/retry_filter.cc (hunk 1 above)
Comment (markdroth) [reply src/core/ext/filters/client_channel/retry_filter.cc:1599]: I'll fix this in a subsequent PR, to avoid delaying this one further.

**Pull request merged**